sklearn/model_selection/partition.py

```diff
@@ -0,0 +1,1129 @@
+        )
+
+    def __len__(self):
+        return int(factorial(self.n) / factorial(self.n - self.p)
+                   / factorial(self.p))
+
+
+class _BaseKFold(with_metaclass(ABCMeta, _PartitionIterator)):
+    """Base class to validate KFold approaches"""
+
+    @abstractmethod
+    def __init__(self, n, n_folds, shuffle, random_state):
+        super(_BaseKFold, self).__init__(n)
+
+        if abs(n_folds - int(n_folds)) >= np.finfo('f').eps:
+            raise ValueError("n_folds must be an integer")
+        self.n_folds = n_folds = int(n_folds)
+
+        if n_folds <= 1:
+            raise ValueError(
+                "k-fold cross validation requires at least one"
+                " train / test split by setting n_folds=2 or more,"
+                " got n_folds={0}.".format(n_folds))
+        if n_folds > self.n:
+            raise ValueError(
+                ("Cannot have number of folds n_folds={0} greater"
+                 " than the number of samples: {1}.").format(n_folds, n))
+
+        if not isinstance(shuffle, bool):
+            raise TypeError("shuffle must be True or False;"
+                            " got {0}".format(shuffle))
+        self.shuffle = shuffle
+        self.random_state = random_state
+
+
+class KFold(_BaseKFold):
+    """K-Folds cross validation iterator.
+
+    Provides train/test indices to split data in train test sets. Split
+    dataset into k consecutive folds (without shuffling).
+
+    Each fold is then used a validation set once while the k - 1 remaining
+    fold form the training set.
+
+    Read more in the :ref:`User Guide <partition>`.
+
+    Parameters
+    ----------
+    n : int
+        Total number of elements.
+
+    n_folds : int, default=3
+        Number of folds. Must be at least 2.
+
+    shuffle : boolean, optional
+        Whether to shuffle the data before splitting into batches.
+
+    random_state : None, int or RandomState
+        Pseudo-random number generator state used for random
+        sampling. If None, use default numpy RNG for shuffling
+
+    Examples
+    --------
+    >>> from sklearn.model_selection import partition
+    >>> X = np.array([[1, 2], [3, 4], [1, 2], [3, 4]])
+    >>> y = np.array([1, 2, 3, 4])
+    >>> kf = partition.KFold(4, n_folds=2)
+    >>> len(kf)
+    2
+    >>> print(kf)  # doctest: +NORMALIZE_WHITESPACE
+    sklearn.partition.KFold(n=4, n_folds=2, shuffle=False,
+                                   random_state=None)
+    >>> for train_index, test_index in kf:
+    ...    print("TRAIN:", train_index, "TEST:", test_index)
+    ...    X_train, X_test = X[train_index], X[test_index]
+    ...    y_train, y_test = y[train_index], y[test_index]
+    TRAIN: [2 3] TEST: [0 1]
+    TRAIN: [0 1] TEST: [2 3]
+
+    Notes
+    -----
+    The first n % n_folds folds have size n // n_folds + 1, other folds have
+    size n // n_folds.
+
+    See also
+    --------
+    StratifiedKFold: take label information into account to avoid building
```

Comment:
This syntax for `see also` doesn't render well, see [here](
http://scikit-learn.org/stable/modules/generated/sklearn.cross_validation.KFold.html#sklearn.cross_validation.KFold)

Maybe indenting the lines below fixes it? Or backslash-escaping the end of lines. Or simply keeping the description on one line: 
`StratifiedKFold: for imbalanced class distributions`